nflow-rest-api-spring-web/src/main/java/io/nflow/rest/v1/springweb/StatisticsResource.java

```diff
@@ -34,22 +37,25 @@
   private StatisticsService statisticsService;
   @Inject
   private StatisticsConverter statisticsConverter;
+  @Inject
+  private SchedulerService scheduler;
 
   @GetMapping
   @ApiOperation(value = "Get executor group statistics", response = StatisticsResponse.class, notes = "Returns counts of queued and executing workflow instances.")
-  public ResponseEntity<?> queryStatistics() {
-    return handleExceptions(() -> ok(statisticsConverter.convert(statisticsService.getStatistics())));
+  public Mono<ResponseEntity<?>> queryStatistics() {
+    return handleExceptions(
```

Comment:
Could this wrapBlocking be moved to handleExceptions? Or create handleExceptionsAndWrap helper for it?

Reply:
It could for some of the endpoints but not all. Everything endpoint doesn't have everything blocking, and there are others where not all parts of the call are blocking.

I'm not also very sure handling exceptions and using a thread pool is conceptually similar enough to make a method doing both sound? To me it sounds kinda like a hodge podge of "doGoodThings". Sure it's shorter, but is it more understandable?

But I can do it if you think it is sound in the context of the rest of the codebase :)

Reply:
Well it just seemed to repeat quite often, so I was thinking if it could be moved to base class, but maybe it is not worth it. Another repeating thing is scheduler service field in many resources, could that be moved to base class? But that is not mandatory either, just an idea.

Reply:
Moved to the base class this `wrapBlocking` now in all of the XResources. Also using constructor injection with `@Inject` systematically.